step1/vanila-todo/TodoModel.ts

```diff
@@ -0,0 +1,110 @@
+import TodoView from "./TodoView";
+import TodoController from "./TodoController";
+
+type todoItem = {
+  id: number;
+  text: string;
+  completed: boolean;
+};
+type stateProp = {
+  selected: number;
+  todoList: todoItem[];
+};
+export default class TodoModel {
+  currentInputData: string;
+  editInputData: string;
+  Index: number;
+  state: stateProp | null;
+  view: TodoView;
+  controller: TodoController;
```

Comment:
```suggestion
  private currentInputData: string;
  private editInputData: string;
  private Index: number;
  private state: stateProp | null;
  private view: TodoView;
  private controller: TodoController;
```
이렇게 접근제한자를 지정해주세요 ㅎㅎ
객체지향 패러다임에서는 멤버변수를 외부에 노출시키지 않는게 원칙이랍니다!
현재 클래스에서 public 으로 사용되고 있는 멤버변수가 있다면 설계가 잘못된거라고 보시면 좋을 것 같아요.